### ISSUE TITLE
all tests fail on ruby 2.5 during debian package build

### ISSUE THREAD

**Issue by lelutin** [piotrmurach/necromancer#8]
Hello,

I'm working on packaging necromancer for debian and I got errors when the tests ran. All of the tests fail with the same error:

~~~
RUBYLIB=. GEM_PATH=debian/ruby-necromancer/usr/share/rubygems-integration/all:/home/gabster/.gem/ruby/2.5.0:/var/lib/gems/2.5.0:/usr/lib/ruby/gems/2.5.0:/usr/share/rubygems-integration/2.5.0:/usr/share/rubygems-integration/all:/usr/lib/x86_64-linux-gnu/rubygems-integration/2.5.0 ruby2.5 -S rake -f debian/ruby-tests.rake
/usr/bin/ruby2.5 /usr/bin/rspec --pattern ./spec/\*\*/\*_spec.rb --format documentation

An error occurred while loading ./spec/unit/can_spec.rb.
Failure/Error:
  RSpec.describe Necromancer, 'can?' do
    it "checks if conversion is possible" do
      converter = described_class.new
      expect(converter.can?(:string, :integer)).to eq(true)
      expect(converter.can?(:unknown, :integer)).to eq(false)
    end
  end

NameError:
  uninitialized constant Necromancer
# ./spec/unit/can_spec.rb:3:in `<top (required)>'
~~~

runnin on debian sid in a chroot jail (vanilla debian) with prerequisites for the gem installed by the package build process.

am I maybe missing something in order for the tests to be able to run?

**Comment by lelutin** [comment]
github won't let me attach the file..
here's a link to the full build log which might give more insight about what was installed in the vanilla system:

https://paste.debian.net/1129981/

the paste will expire 90 days from today

**Comment by piotrmurach** [comment]
Hey Gabriel 👋 

Since the specs cannot load `necromancer` it means that the `spec_helper.rb` file is not being loaded. The loading is done by the `.rspec` file which is not included in the gemspec file - my bad. Actually, I stopped including test files in my gem releases. They bloat packages and have no place in released gems. Are you building the gem from the tar? Can you try building from the GitHub source code?

**Comment by lelutin** [comment]
@piotrmurach thanks! someone in the ruby packaging team also pointed the same to me today.

if I add just the `.rspec` file back in (with a package patch), then the tests run without errors.

I'm building the package using `gem2deb`, which is a helper script that downloads the pre-built gem (I believe from rubygems.org) and then extracts its contents and slaps debian-specific files on top with some pre-populated values so that it's easy to jumpstart building a package.

I guess I could build from github source instead. I'll try and get some information about how to do this in the future.

**Comment by piotrmurach** [comment]
I will update this gem to exclude tests as this is the choice I've made on all other of my gems. So in a way, this may actually solve your problem - no tests to run. 

The `gem2deb` seems a bit dated but if it does the job then that's cool. 

I'd be curious to learn if you can succeed with the GitHub source. I want to make sure that you can build a Debian package. A lot of my `tty` gems are distributed on Linux distros like ArchLinux etc. So I do appreciate the work that's involved and I'm happy for the gems to be distributed via different package managers.

**Comment by lelutin** [comment]
heh actually running tests during builds is interesting for debian packages, since builds are all done in CI. this way it's easier to spot if a library change or a ruby version change breaks functionality somewhere.

The debian file (`debian/watch`) that instructs tools where to get upstream code from for all tty toolkit libraries so that code is now fetched from github. I've just applied the same change so that the future releases of necromancer also get packaged from the repository code.

I guess, I'll see if the build works well with the next upstream release

**Comment by piotrmurach** [comment]
Released `v0.6.0`.
